apps/dcos_l4lb/src/dcos_l4lb_mesos_poller.erl

```diff
@@ -39,7 +39,9 @@
 -type lkey() :: {key(), riak_dt_orswot}.
 -type backend() :: {
     AgentIP :: inet:ip4_address(),
-    {BackendIP :: inet:ip_address(), BackendPort :: inet:port_number() }
+    {BackendIP :: inet:ip_address(),
+     BackendPort :: inet:port_number(),
+     BackendWeight :: inet:port_number() }
```

Comment:
Maybe `non_neg_integer()` is more suitable that `inet:port_number()` for weight type in this case?

Reply:
The max value for weight is 65535 (16bit integer). I just used `inet:port_number` so not to have to write a type just for this.

Reply:
You should be able to just use `non_neg_integer` here - no need to write a new type for this one.

Reply:
@orlandohohmeier but that would allow invalid larger values than 65535, which are invalid weights. Whereas [`inet:port_number`](https://github.com/erlang/otp/blob/c5cb5ff89b5667b2e17946ed54b5f053a48e55c5/lib/kernel/src/inet.erl#L103) is already restricted appropriately.

Reply:
gotcha, I'd vote for a new type then.

Reply:
Thanks for your explanation.